src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs

```diff
@@ -183,5 +184,59 @@ public void Dispose()
             LogManager.Configuration.LoggingRules.Clear();
             LogManager.ReconfigExistingLoggers();
         }
+
+        /// <summary>
+        /// By default, logs are disabled when using <see cref="Create(string)"/> or <see cref="Create(object, string)"/> methods,
+        /// so using this fluent method, the caller can enable the logs at will.
+        /// </summary>
+        /// <returns>Current <see cref="NodeBuilder"/> instance, used for fluent API style</returns>
+        /// <example>
+        /// //default use (without logs)
+        /// using (NodeBuilder builder = NodeBuilder.Create(this))
+        /// {
+        ///     //your test code here
+        /// }
+        ///
+        /// //with logs enabled
+        /// using (NodeBuilder builder = NodeBuilder.Create(this).WithLogsEnabled())
+        /// {
+        ///     //your test code here
+        /// }
+        /// </example>
+        public NodeBuilder WithLogsEnabled()
+        {
+            if (!LogManager.IsLoggingEnabled())
+            {
+                // NLog Enable/Disable logging is based on internal counter, so to be sure to have enabled logs, must to
+                // keep calling EnableLogging until IsLoggingEnabled returns true.
+                do
+                {
+                    LogManager.EnableLogging();
+                } while (!LogManager.IsLoggingEnabled());
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// If logs have been enabled by calling WithLogsEnabled you can disable it manually by colling this method.
+        /// If the test is running within an "using block" where the nodebuilder is created, without using <see cref="WithLogsEnabled"/>,
+        /// you shouldn't need to call this method.
+        /// </summary>
+        /// <returns>Current <see cref="NodeBuilder"/> instance, used for fluent API style</returns>
+        public NodeBuilder WithLogsDisabled()
+        {
+            if (LogManager.IsLoggingEnabled())
+            {
+                // NLog Enable/Disable logging is based on internal counter, so to be sure to have enabled logs, must to
+                // keep calling DisableLogging until IsLoggingEnabled returns false.
```

Comment:
See above comment on this.

Reply:
i found that I had to change "to be sure to have enabled" to "to be sure to have *disabled*" in this context too.